### DISCUSSION TITLE
past_due subscriptions and automatic retry payments in Stripe resulted in multiple subscriptions for a user

### DISCUSSION THREAD

**Discussion by mpvosseller** [pay-rails/pay#1040] [General] [answered]
We discovered a flow in our app that allowed a user to have two paid / active Stripe subscriptions at the same time (which our app wants to prevent):

- user buys a monthly subscription
- at some point a renewal charge fails because the credit card is expired
- the subscription status is set to `past_due`
- the user is now no longer considered "subscribed" because `on_trial_or_subscribed?` returns `false`
- a few days later the user notices they are unsubscribed and purchase another subscription. We currently allow a user to buy a subscription if `on_trial_or_subscribed?` returns `false`). The user enters a new credit card.
- the next day Stripe re-tries a charge for the original `past_due` subscription (Stripe's default settings are to retry charges for 15 days). The charge works because the user has a new working credit card on file.
- The user now has two active paid subscriptions.

The most obvious solution is to prevent the user from buying a new subscription if they are already subscribed (`user.payment_processor.on_trial_or_subscribed?`) OR if they have  one that is in the `past_due` status (`user.payment_processor.subscription.past_due?`)

Wondering if this matches what folks are currently doing or if anyone handles this another way.

Is it worth adding a note about this case in the [documentation on `on_trial_or_subscribed?`](https://github.com/pay-rails/pay/blob/main/docs/6_subscriptions.md#checking-customer-trial-or-subscribed-status)?

Happy to do that although I'm not sure how this works with other processors.

**Comment by excid3** [comment, marked as answer]
The joy of payments is in all these little edge cases!

I would consider them not subscribed when they're past due and display a banner that they need to update their payment method to continue using the service. Once it's cancelled or unpaid on Stripe, then you can consider them fully inactive.

**Reply by mpvosseller** [reply]
Thanks for confirming @excid3. That matches what I was thinking.
